app/models/concerns/filterable.rb

```diff
@@ -0,0 +1,13 @@
+module Filterable
+  extend ActiveSupport::Concern
+
+  module ClassMethods
+    def filter(filtering_params)
+      results = self.where(nil)
```

Comment:
it's great i think. took a bit of effort to understand it but i see how this will allow us to add new filtering options without increasing code complexity :+1: